### DISCUSSION TITLE
Multiple GPU support

### DISCUSSION THREAD

**Discussion by v1ckxy** [mathoudebine/turing-smart-screen-python#381] [Q&A]
Hello!
I've been looking around github and haven't found it... so I'm posting the question here.
Is there any way to configure a theme to display multiple GPUs /* or to specify a specific gpu */?

I've looked at the code (sensors_librehardwaremonitor.py and sensors_python.py) and it's not clear to me how to specify a specific card.

The system where I plan to use the display has 2 (and soon 3) NV cards and 1 integrated AMD.

Regards and thank you for this great work.

**Comment by mathoudebine** [comment]
Hello @v1ckxy!
For now the program has not been developed to support multiple GPUs 😞 
This doesn't mean it's impossible though! But it needs big changes on sensors_librehardwaremonitor.py / sensors_python.py and also on the theme parser to be able to specify multiple GPUs inside `theme.yaml` files

To use a specific GPU:

For sensors_librehardwaremonitor.py:
At first start, the program will log all supported GPUs:
`07/11/2023 19:46:08 [INFO] Found AMD GPU: Radeon ...`
To use a specific GPU from the list, copy the GPU name then edit this line:
https://github.com/mathoudebine/turing-smart-screen-python/blob/d44505f03ab7f35cbb9f7eb9bf3bed86d182d458/library/sensors/sensors_librehardwaremonitor.py#L160
to:
`         return "Radeon ..."`

For sensors_python.py:
For Nvidia cards the program uses all available Nvidia cards sensors from `GPUtil.getGPUs()` and returns an average:
https://github.com/mathoudebine/turing-smart-screen-python/blob/d44505f03ab7f35cbb9f7eb9bf3bed86d182d458/library/sensors/sensors_python.py#L139-L149
You would have to edit the code to use only the GPU you want from the `GPUtil.getGPUs()` method

**Comment by v1ckxy** [comment]
Thanks! I will have a look at those parts of the code.

Maybe I can just implement a foreach somewhere and then throw GPU0, 1, 2...
